include/SFML/Window/WindowBase.inl

```diff
@@ -53,20 +54,56 @@ struct DelayOverloadResolution
     {
     }
 };
+
+// Wrapper providing a callable type suitable as OverloadSet base type for different kinds of handlers.
+// By default, we derive from the handler type and inherit its call operators.
+template <typename Handler>
+struct Caller : Handler
+{
+    using Handler::operator();
+};
+
+// Inheritance is not possible with reference types.
+// In this case, we capture the reference and forward arguments to it.
+template <typename Handler>
+struct Caller<Handler&>
+{
+    // Use SFINAE so that the call operator exists only for arguments the captured handler accepts
+    template <typename Argument, std::enable_if_t<std::is_invocable_v<Handler&, Argument>, int> = 0>
+    decltype(auto) operator()(Argument&& argument)
+    {
+        return handler(std::forward<Argument>(argument));
+    }
+    Handler& handler;
+};
+
+// Inheritance is not possible with function pointers either.
+// In this case, we capture the function pointer to call it.
+template <typename Return, typename Argument>
+struct Caller<Return (*)(Argument)>
+{
+    Return operator()(Argument&& argument)
+    {
+        return function(std::forward<Argument>(argument));
+    }
+    Return (*function)(Argument);
+};
 } // namespace priv
 
 
 ////////////////////////////////////////////////////////////
-template <typename... Ts>
-void WindowBase::handleEvents(Ts&&... handlers) // NOLINT(cppcoreguidelines-missing-std-forward)
+template <typename... Handlers>
+void WindowBase::handleEvents(Handlers&&... handlers)
 {
-    static_assert(sizeof...(Ts) > 0, "Must provide at least one handler");
+    static_assert(sizeof...(Handlers) > 0, "Must provide at least one handler");
+    static_assert((Event::isEventHandler<Handlers> && ...), "Handlers must accept at least one subtype of sf::Event");
 
     // Disable misc-const-correctness for this line since clang-tidy
     // complains about it even though the code would become uncompilable
 
     // NOLINTNEXTLINE(misc-const-correctness)
```

Comment:
I searched https://github.com/SFML/SFML/pull/3139 for an explanation of this `NOLINTNEXTLINE` line but couldn't find an explanation. Do you think we need to keep this? I'm not sure why `overloadSet` would need to be mutable.

Reply:
We don't want `overloadSet` to be const because that would make it impossible to use mutable handlers, for which the call operator is not const.
I though clang-tidy would suggest to make this object const but I don't reproduce locally, so I will remove the comment and check what CI finds.

Reply:
I removed that `NOLINT` and re-ran clang-tidy without errors so I think we're good.

Reply:
CI is not happy:

```
/home/runner/work/SFML/SFML/include/SFML/Window/WindowBase.inl:101:5: error: variable 'overloadSet' of type [...] can be declared 'const' [misc-const-correctness,-warnings-as-errors]
```

https://github.com/SFML/SFML/actions/runs/12978770634/job/36193726247?pr=3399#step:8:13

Reply:
A workaround is to use `auto`. clang-tidy does not recommend adding `const` to any local variables declared with `auto`.

[Ask me how I know :)](https://github.com/llvm/llvm-project/issues/60789)

Reply:
I would rather add a NOLINT comment than rely on a clang-tidy bug.
I will reset my branch to the commit with the comment.

Reply:
The comment is still incorrect though. Adding `const` does not make this code uncompilable as the comment claims.

Reply:
I changed "uncompilable" into "incorrect".